staff/judith-gongora/pre-course/bingo/bingo.js

```diff
@@ -0,0 +1,224 @@
+//Bingo Proyecto Tema 3
+var numerosdados = [] //Variable global Array para ir guardando números que han salido ya
+var linea = false //Variable global para comprobar si hay linea y que no vuelva a darnosla
+var turnos = 0 //Variable global que suma los turnos
+var bingoCard = [] //Variable global array para generar los numeros (objetos)
+var primero = true
+
+function newCard(){
+  primero = true
+  numerosdados = [] //Instanciamos de nuevo para retear el Array
+  turnos = 0 //Cada vez que generamos un cartón reseteamos la variable turnos
+  linea = false //Cada vez que generamos un cartón reseteamos la variable linea
+  bingoCard = [ //Generamos los objetos numeros con todo a falso (matched) y el número aleatorio con la función randomNumber
+    {number: randomNumber(), matched: false},
+    {number: randomNumber(), matched: false},
+    {number: randomNumber(), matched: false},
+    {number: randomNumber(), matched: false},
+    {number: randomNumber(), matched: false},
+    {number: randomNumber(), matched: false},
+    {number: randomNumber(), matched: false},
+    {number: randomNumber(), matched: false},
+    {number: randomNumber(), matched: false},
+    {number: randomNumber(), matched: false},
+    {number: randomNumber(), matched: false},
+    {number: randomNumber(), matched: false},
+    {number: randomNumber(), matched: false},
+    {number: randomNumber(), matched: false},
+    {number: randomNumber(), matched: false}
+  ]
+  var comprobar = []  //Array para almacenar numeros y poder comprobar que no se repiten en el cartón
+  bingoCard.forEach(function(obj){ //Recorremos Array de objetos
+      comprobar.push(obj.number) //Añadimos en Array comprobar los números generados
+  })
+  comprobar.sort() //Ordenamos Array descendiente
+  var comprobacion = false //Boleano para comprobar si existe coincidencia de número en el cartón
+  for(i = 1; i < comprobar.length; i++) { // recorremos todo el Array ordenado
+    if(comprobar[i] == comprobar[i - 1]) { // Comparamos si el contenido de la posicion i es igual a su anterior
+      comprobacion = true //Se repite por lo tanto cambiamos boleano
+    }
+  }
+  if (comprobacion){
+    newCard() //Si se repite volvemos a generar un cartón, este no sirve
+  }else{ // Si no se repite
+    console.log("Aquí tienes tu cartón para empezar a jugar.\n")
+    showCard() //Mostramos el cartón
+  }
+}
+
+function randomNumber(){ //Funcioó que nos devuelve un número aleatorio entre 1 y 99
+  return Math.floor(Math.random() * 99) + 1
+}
```

Comment:
the output number can be repeated

Reply:
In the function numeroNuevo the numbers are compared.

Reply:
my bad then :(